src/common/types/core.js

```diff
@@ -43,14 +135,252 @@ type SelectorItem = {
+    can_upload?: boolean,
+};
+
+type BoxItemVersionPermission = {
+    can_delete?: boolean,
+    can_download?: boolean,
+    can_preview?: boolean,
+    can_upload?: boolean,
+};
+
+type BoxItemVersionRetentionPolicy = {
+    disposition_action: typeof VERSION_RETENTION_DELETE_ACTION | typeof VERSION_RETENTION_REMOVE_ACTION,
+    id: string,
+    policy_name: string,
+    retention_length: typeof VERSION_RETENTION_INDEFINITE | string, // length in days
+    type: 'retention_policy',
+};
+
+type BoxItemVersionRetention = {
+    applied_at: string,
+    disposition_at: string,
+    id: string,
+    type: 'file_version_retention',
+    winning_retention_policy: BoxItemVersionRetentionPolicy,
+};
+
+type BoxItemVersion = {
+    authenticated_download_url?: string,
+    collaborators?: Object,
+    created_at: string,
+    extension?: string,
+    id: string,
+    is_download_available?: boolean,
+    modified_at?: string,
+    modified_by: ?User,
+    name?: string,
+    permissions?: BoxItemVersionPermission,
+    restored_at?: string,
+    restored_by?: ?User,
+    retention?: BoxItemVersionRetention,
+    sha1?: string,
+    size?: number,
+    trashed_at: ?string,
+    trashed_by?: ?User,
+    type: 'file_version',
+    version_end?: number,
+    version_number: string,
+    version_promoted?: string,
+    version_start?: number,
+    versions?: Array<BoxItemVersion>,
+};
+
+type BoxPathCollection = {
+    entries: Array<Crumb>,
+    total_count: number,
+};
+
+type FileRepresentation = {
+    content?: {
+        url_template: string,
+    },
+    properties?: {
+        dimensions: string,
+        paged: string,
+        thumb: string,
+    },
+    representation?: string,
+    status: {
+        state: string,
+    },
+};
+
+type FileRepresentationResponse = {
+    entries: Array<FileRepresentation>,
+};
+
+type BoxItem = {
+    allowed_shared_link_access_levels?: Array<Access>,
+    authenticated_download_url?: string,
+    content_created_at?: string,
+    content_modified_at?: string,
+    created_at?: string,
+    created_by?: User,
+    description?: string,
+    download_url?: string,
+    extension?: string,
+    file_version?: BoxItemVersion,
+    has_collaborations?: boolean,
+    id: string,
+    interacted_at?: string,
+    is_download_available?: boolean,
+    is_externally_owned?: boolean,
+    item_collection?: BoxItemCollection, // eslint-disable-line no-use-before-define
+    metadata?: MetadataType,
+    modified_at?: string,
+    modified_by?: User,
+    name?: string,
+    owned_by?: User,
+    parent?: BoxItem,
+    path_collection?: BoxPathCollection,
+    permissions?: BoxItemPermission,
+    representations?: FileRepresentationResponse,
+    restored_from?: BoxItemVersion,
+    selected?: boolean,
+    shared_link?: SharedLink,
+    size?: number,
+    thumbnailUrl?: ?string,
+    type?: ItemType,
+    url?: string,
+    version_limit?: ?number,
+    version_number?: string,
+};
+
+type BoxItemCollection = {
```

Comment:
We can probably move this above `BoxItem` and remove the comment on line 244

Reply:
@booc0mtaco they each depend on each other, so neither order works entirely.